src/main/java/io/stargate/sgv2/jsonapi/service/operation/tables/WriteableTableRowBuilder.java

```diff
@@ -0,0 +1,243 @@
+import io.stargate.sgv2.jsonapi.exception.DocumentException;
+import io.stargate.sgv2.jsonapi.exception.ServerException;
+import io.stargate.sgv2.jsonapi.exception.catchable.MissingJSONCodecException;
+import io.stargate.sgv2.jsonapi.exception.catchable.ToCQLCodecException;
+import io.stargate.sgv2.jsonapi.exception.catchable.UnknownColumnException;
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.TableSchemaObject;
+import io.stargate.sgv2.jsonapi.service.operation.filters.table.codecs.*;
+import io.stargate.sgv2.jsonapi.service.shredding.*;
+import io.stargate.sgv2.jsonapi.service.shredding.collections.JsonPath;
+import io.stargate.sgv2.jsonapi.service.shredding.tables.WriteableTableRow;
+import java.util.*;
+
+/**
+ * Builds a {@link WriteableTableRow} from a {@link JsonNamedValueContainer}.
+ *
+ * <p>This class does all the checks that the columns and data we are trying to insert are valid, it
+ * enforces the rules that {@link WriteableTableRow} has to be valid.
+ */
+public class WriteableTableRowBuilder {
+
+  private final TableSchemaObject tableSchemaObject;
+  private final TableMetadata tableMetadata;
+  private final JSONCodecRegistry codecRegistry;
+
+  /**
+   * Creates a new instance of the {@link WriteableTableRowBuilder}
+   *
+   * @param tableSchemaObject the table schema to validate the {@link JsonNamedValueContainer}
+   *     against.
+   * @param codecRegistry the codec registry to use to convert the raw JSON values to CQL values.
+   */
+  public WriteableTableRowBuilder(
+      TableSchemaObject tableSchemaObject, JSONCodecRegistry codecRegistry) {
+    this.tableSchemaObject =
+        Objects.requireNonNull(tableSchemaObject, "tableSchemaObject cannot be null");
+    this.tableMetadata = tableSchemaObject.tableMetadata;
+    this.codecRegistry = Objects.requireNonNull(codecRegistry, "codecRegistry cannot be null");
+  }
+
+  /**
+   * The {@link TableSchemaObject} that this builder is using to validate the {@link
+   * JsonNamedValueContainer}.
+   */
+  public TableSchemaObject tableSchemaObject() {
+    return tableSchemaObject;
+  }
+
+  /**
+   * Builds a {@link WriteableTableRow} from a {@link JsonNamedValueContainer}, validating schema
+   * and data using the state of the builder.
+   *
+   * @param source the {@link JsonNamedValueContainer} to build the {@link WriteableTableRow} from.
+   * @return A valid {@link WriteableTableRow} that can be inserted.
+   */
+  public WriteableTableRow build(JsonNamedValueContainer source) {
+
+    // To be extra safe, create the CQLIdentifier for each JsonNamedValue and link it back
+    // to the entry from the container, we will use this later when we convert the JSON value with
+    // the codec
+    // NOTE: this is where we DOUBLE QUOTE every raw field name to make sure it is a valid CQL
+    // identifier - see cqlIdentifierToJsonValue()
+    Map<CqlIdentifier, JsonNamedValue> cqlIdentifierToJsonValue = new HashMap<>();
```

Comment:
Ok I am still/again confused: if this is for Tables, shouldn't we have `CqlNamedValue` and NOT `JsonNamedValue`?

Reply:
the cqlIdentifierToJsonValue map uses JsonNamedValue because it is processing a JsonNamedValueContainer and the first step I take is to convert the JSON name to a CQL identifier and map that back to the JSONNamedValue I did that for.